### DISCUSSION TITLE
Is this project dead?

### DISCUSSION THREAD

**Discussion by gitizenme** [dotnet/command-line-api#2332] [General]
We're looking for a CLI framework for a new set of applications we're building that will have a 10+ year lifecycle. We enjoy coding with this package and would like to move forward with it. 
However, the silence from the project owners on GA and the lack of a roadmap raises some concerns. 

For the product owners, what is the long-term viability of this framework?

**Comment by elgonzo** [comment]
This released today/yesterday: https://github.com/dotnet/command-line-api/issues/2338

**Reply by dragon7307** [reply]
The nuget package has gone stale since 2022, any plans to update that?

**Comment by calacayir** [comment]
Did you hear about DotMake.CommandLine which makes working with System.CommandLine very easy? You just write your cli commands as simple classes and it wires it up for you by using a source generator. This way you can focus on logic of your complex cli app and not parsing.
https://github.com/dotmake-build/command-line

**Comment by KalleOlaviNiemitalo** [comment]
If one is worrying about whether System.CommandLine will be maintained and supported in the future, then I don't think one can address that concern by adding a dependency on another package that still depends on System.CommandLine.

**Reply by calacayir** [reply]
They still release daily builds and we depend on them (bundle them), if it makes a difference.

**Reply by calacayir** [reply]
And the main idea of my project is the source generator, which means it can be switched to any CLI parser in the future, e.g. Spectre.Console.Cli but I am not fond of it, visual part is great but I still think System.CommandLine is a better parser and it has better model. Daily builds of System.CommandLine work perfectly and I don't think it will be dead soon.

**Comment by brpelton238** [comment]
There is a discussion here that is about a year old - sounds like they ~are~ _were_ doing some rework on the project:
https://www.youtube.com/watch?v=SzpFCVg3JbA

**Comment by elgonzo** [comment]
@brpelton238 
> sounds like they are doing some rework on the project

Considering that there is no visible activity in the powderhouse branch since about 9 months, and that the discussions by the respective members of the .NET team here in the Github repo stopped also around the same time, they must be remarkably good at hiding their activities regarding reworking of the project....

**Reply by brpelton238** [reply]
Thanks, I didn't realize the work stopped.  Ugh. 
Spectre Console might be the way.

**Reply by elgonzo** [reply]
Lol, just one day after our exchange, this happened: https://github.com/dotnet/command-line-api/issues/2576

**Reply by jonsequitur** [reply]
It wasn't easy to 🤐
